lib/teamsnap.rb

```diff
@@ -137,13 +138,40 @@ def init(opts = {})
       collection = TeamSnap.run_init(:get, "/", {})
 
       classes = []
-      client.in_parallel do
-        classes = collection
-          .fetch(:links)
-          .map { |link| classify_rel(link) }
+
+      schema = collection
+        .fetch(:links) { {} }
+        .find { |link| link[:rel] == "schemas" } || {}
+
+      if schema[:href]
+        href_to_rel = collection
+          .fetch(:links) { {} }
+          .reject { |link| EXCLUDED_RELS.include?(link[:rel]) }
+          .map { |link| [link[:href], link[:rel]]}
+          .to_h
+
+        resp = client.get(schema[:href])
+        if resp.status == 200
+          collections = Oj.load(resp.body)
+          classes = collections.map { |collection|
+            col = collection.fetch(:collection) { {} }
+            if rel = href_to_rel[col[:href]]
+              create_collection_class(rel, col[:href], nil, col)
+            end
+          }
           .compact
+        else
+          error_message = TeamSnap.parse_error(resp)
+          raise TeamSnap::Error.new(error_message)
+        end
+      else
+        client.in_parallel do
+          classes = collection
+            .fetch(:links) {}
```

Comment:
this would give a default of nil... you might want to have it return an empty array to prevent an error?